wp-content/civi-extensions/goonjcustom/Civi/CollectionCampService.php

```diff
@@ -1354,4 +1355,53 @@ public function autofillReceiptFrom($formName, &$form) {
     }
   }
 
+  /**
+   *
+   */
+  public static function updateCampaignForCollectionSourceContribution(string $op, string $objectName, $objectId, &$objectRef) {
+    if ($objectName != 'Contribution' || $objectId === NULL) {
+      return;
+    }
+
+    $contributionId = $objectRef['id'] ?? NULL;
+    if (!$contributionId) {
+      return;
+    }
```

Comment:
I don't understand the need for this. This code is called on the pre hook so we can assume the contribution has not created just yet. Can't we?

Reply:
When I click on the contribute button after filling out the form, this hook runs. Inside the objectRef, I was able to get the contributionId. With the help of that contribution ID, I was fetching out the `collectionSourceId`. Based on that camp ID, I found the campaign associated with that camp.